comment.js

```diff
@@ -1,34 +1,80 @@
 import {getCommentPrefix} from './identifier';
 
-export async function findMatchingComment({octokit, owner, repo, issue_number, identifier}) {
-  const matchingComments = await findMatchingComments({octokit, owner, repo, issue_number, identifier});
+const issueCommenter = {
+    createComment: async function(octokit, context, data) {
+        data.issue_number = context;
+        await octokit.issues.createComment(data);
+    },
+    deleteComment: async function(octokit, data) {
+        data.issue_number = context;
+        await octokit.issues.deleteComment(data);
+    },
+    updateComment: async function(octokit, data) {
+        data.issue_number = context;
+        await octokit.issues.updateComment(data);
+    },
+    listComments: async function(octokit, context, data) {
+        data.issue_number = context;
+        await octokit.issues.listComments(data);
+    }
+}
```

Comment:
Instead of storing just the methods in a plain object, we will move towards using [ES6 classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) that will help us store both data (`octokit`, `number`) along with the methods.

After this refactoring, we will be able instantiate the logic like shown below and will use it in other parts of the code base.
```javascript
let commenter = new IssueCommenter({ octokit, owner, repo, number }) // we will be using this in `getCommenterMethod` 

commenter.createComment(message);
commenter.deleteComment(commentId);
commenter.updateComment(commentId, message);
commenter.listComments(opts); // expecting opts.page and opts.per_page to be present for now  
```